roles/icingaweb2/tasks/manage_db.yml

```diff
@@ -0,0 +1,29 @@
+---
+
+- name: Prepare database
+  ansible.builtin.include_tasks: "{{ icingaweb2_db.type }}/prepare_db.yml"
+  when: icingaweb2_db is defined
+
+- name: Import database schema
+  ansible.builtin.include_tasks: "{{ icingaweb2_db.type }}/import_db.yml"
+  when: icingaweb2_db_import_schema | default(False)
+
+- name: Add admin to users list when users is defined
+  ansible.builtin.set_fact:
+    icingaweb2_users: '{{ icingaweb2_users + [{"username": "{{ icingaweb2_admin_username }}", "password": "{{ icingaweb2_admin_password }}", "recreate": "{{ icingaweb2_admin_recreate is defined }}" }]}}'
+  when: icingaweb2_admin_username is defined and icingaweb2_admin_password is defined and icingaweb2_users is defined
+
+- name: Add Icinga web 2 users
+  ansible.builtin.include_tasks: "{{ icingaweb2_db.type }}/users_db.yml"
+  loop: "{{ icingaweb2_users }}"
+  loop_control:
+    loop_var: _users
+  when: icingaweb2_users is defined
+
+- name: Add Icingaweb2 admin 
+  ansible.builtin.include_tasks: "{{ icingaweb2_db.type }}/users_db.yml"
+  loop:
+    - { username: '{{ icingaweb2_admin_username }}', password: '{{ icingaweb2_admin_password }}', recreate: '{{ icingaweb2_admin_recreate is defined }}' }
+  loop_control:
+    loop_var: _users
+  when: icingaweb2_admin_username is defined and icingaweb2_admin_password and icingaweb2_users is undefined
```

Comment:
This can be reduced to only two tasks? 
You are creating the list icingaweb2_users, only if icingaweb2_users is defined and admin_user is defined.
If one of those statements is true you get a list of users right? I know that you probably choose this because if icingaweb2_users is not defined you get an error. 

But if you define icingaweb2_users as empty array. `icingaweb2_users: []` then you can add the admin user and get a list with one user inside. 

In the end you only need to check, are there more than 0 items in the list. Then you can loop over those and create the users.

Reply:
Thanks for the suggestion, now implemented.